options/options.go

```diff
@@ -325,7 +325,7 @@ func (interp *interpreter) interpretFieldOptions(fqn string, fld *descriptorpb.F
 	if index >= 0 {
 		opt := uo[index]
 		optNode := interp.file.OptionNode(opt)
-		if fld.GetExtendee() != "" {
+		if extendee := fld.GetExtendee(); extendee != "" && fld.GetJsonName() != string(opt.GetStringValue()) {
```

Comment:
Now that we want to _use_ the string value here, I think we need to move this below the check below on line 336. If you move it below line 339, you can just use the `name` variable.

Also, using `fld.GetJsonName` here is a bit subtle/confusing since we are actually in the process of _setting_ that value (on line 343). So I think it would help to either (1) add a comment clarifying that the field is initially set to the field's default JSON name which is why the check works or (2) just explicitly compute the default JSON name (by calling `internal.JSONName(fieldName)`) and compare to that instead.

Finally, why add the extra `extendee` var? I think it's a little more concise and just as clear to do:
```go
if fld.GetExtendee() != "" && name != internal.JSONName(fld.GetName()) {
```